FC preparation for int_oo in PyTorch

ezyang opened the pull request:
Summary: This is needed for https://github.com/pytorch/pytorch/pull/127693 . This code is written so it is compatible before and after this PR.

Differential Revision: D58465158

mergennachin commented on exir/sym_util.py:
```diff
@@ -41,17 +41,23 @@ def eval_upper_bound(maybe_symint: Union[int, torch.SymInt]) -> Optional[int]:
     expr = node.expr
     var_range: ValueRanges = bound_sympy(expr, shape_env.var_to_range)
     upper_bound = var_range.upper
+
```
add a comment that this could be deleted when we move forward the pin everywhere

ezyang committed:
FC preparation for int_oo in PyTorch (#3947)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/3947

This is needed for https://github.com/pytorch/pytorch/pull/127693 . This code is written so it is compatible before and after this PR.

Reviewed By: mergennachin

Differential Revision: D58465158
